index.bs

```diff
@@ -180,88 +181,180 @@ public IP address scope is being discussed in
+- The [=UA=] shall be able to detect the revocation of the [=device=]’s certificate.
+- The [=device=] shall be able to refresh the certificate at any time.
+
+### <dfn>Device Discovery</dfn> ### {#req-device-discovery}
+
+As described in [[#uc-02]] and [[#uc-03]], when the [=web service=] does not know the device URL in advance,
+the [=device=] needs to be discovered. That means the discovery process must meet the following requirements:
+- The [=UA=] shall be able to discover the presence of [=device=]s in the same local network.
+- The [=UA=] shall be able to obtain the endpoint URL of the [=device=] that has the scheme `https://` or `wss://`.
+- When the [=UA=] finds multiple [=device=]s, the [=UA=] shall be able to present the selector interface
+    to display the list of the [=device=]s and allow the user to select one specific [=device=].
+
+## Non-Functional Requirements ## {#req-non-funcational-req}
+
+The non-functional requirements are:
+- Security
+- Privacy
+- Availability
+- Scalability
+- User Experience
+
+### Security ### {#req-security}
+
+Following security considerations are important:
+- If the [=device=] uses a TLS server certificate for HTTPS/WSS communication,
+    the certificate issuance and renewal steps shall prevent passive network
+    eavesdroppers from learning information related to the identification or certificate
+    exchanged among the system components (the CA, the [=device=], the [=UA=] and the [=web service=]).
+- If the [=device=] uses a TLS server certificate for HTTPS/WSS communication,
+    the certificate issuance and renewal steps shall prevent active network attackers
+    from impersonating a [=device=] and observing or altering data intended for the [=CA=] or for the [=UA=].
+- If the [=device=] does not use a TLS server certificate, alternate steps for the [=UA=]
+    to initiate a TLS session to the [=device=] (e.g., sharing and using a PSK or a RPK)
+    shall prevent passive network eavesdroppers from getting the PSK or the private key of the RPK.
+- If the [=device=] does not use a TLS server certificate, alternate steps for the [=UA=]
+    to initiate a TLS session to the [=device=] (e.g., sharing and using a PSK or a RPK)
+    shall prevent active network attackers from impersonating a [=device=] and observing or
+    altering the keys for the [=UA=].
+- [=Device identification=]  (e.g., the pairing method mentioned above) shall prevent
+    passive network eavesdroppers from learning information related to identification.
+- [=Device identification=] and [=device discovery=]  (if the [=UA=] supports it) shall prevent
+    active network attackers from impersonating a [=device=] and observing or altering
+    data intended for the [=CA=] or for the [=UA=].
+
+### Privacy ### {#req-privacy}
+
+The user privacy shall be preserved against the [=web service=] especially when the user
+allows the [=web service=] to access the [=device=] for private use.  In such scenarios,
```

Comment:
Which is what you really mean, web service or web app (front end of the web service)?

Reply:
Sorry. what I meant was 'frontend of the web service'.